ESCOMP/POP2-CESM
# Langmuir turbulence induced entrainment

### Description of changes:

This pull request enables the parameterization of Langmuir turbulence induced entrainment by [Li and Fox-Kemper (2017)](https://doi.org/10.1175/JPO-D-17-0085.1) via CVMix. It includes the following changes:
 - Point the CVMix repository to v0.97b-beta and update the CVMix interface in POP. A new interface of Langmuir turbulence parameterizations in CVMix was introduced in CVMix v0.94b-beta, allowing more options including Langmuir induced entrainment. See [this PR](https://github.com/CVMix/CVMix-src/pull/74) for more information.
- Get the surface layer averaged Langmuir number `LaSL` from WAVEWATCH III for Langmuir induced entrainment.

### Testing:
 
Test case/suite: See below
Test status: climate changing

Fixes [POP2 Github issue #]: N/A

User interface (namelist or namelist defaults) changes?
- This new option is enabled by setting `langmuir_opt = 'lf17'` in POP's namelist.


I have tested these change in the following two steps using a JRA55-do forced G-case in CESM (tag cesm2_2_beta04) running for 60 years (looking at mean mixed layer depth of the last 30 years). Here the control run is using default values for the namelist. Testing with a longer simulation is underway. 

1. Update the CVMix repository to v0.97b-beta but do not turn on Langmuir induced entrainment  (`langmuir_opt = 'vr12-ma'`)
2. Turn on the Langmuir induced entrainment (`langmuir_opt = 'lf17'`)

The first step is not bit-for-bit, as described in [this PR](https://github.com/CVMix/CVMix-src/pull/74). But it gives essentially the same mean climate, with small changes in some coastal regions where the boundary layer is very shallow. See the figure below. It's likely due to changes in the order of some operations in CVMix introduced in v0.94b-beta. Note that the panels below plot the summer/winter mean mixed layer depth in both hemispheres together, as in [Li and Fox-Kemper (2017)](https://doi.org/10.1175/JPO-D-17-0085.1).
![20200515_Entrainment_CESM2p2 001](https://user-images.githubusercontent.com/12438579/82067374-e08b4480-968d-11ea-9c62-d646bf640bf8.jpeg)
 
The second step is climate changing, making the mixed layer deeper in the extratropical regions, as expected in [Li and Fox-Kemper (2017)](https://doi.org/10.1175/JPO-D-17-0085.1). Note the different scale in the color bar of the percentage changes.
![20200515_Entrainment_CESM2p2 002](https://user-images.githubusercontent.com/12438579/82067650-3d86fa80-968e-11ea-8239-80f8c3a64286.jpeg)
